tuftool/src/add_role.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::datetime::parse_datetime;
+use crate::error::{self, Result};
+use crate::source::parse_key_source;
+use chrono::{DateTime, Utc};
+use snafu::ResultExt;
+use std::fs::File;
+use std::num::NonZeroU64;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tempfile::tempdir;
+use tough::editor::RepositoryEditor;
+use tough::http::HttpTransport;
+use tough::key_source::KeySource;
+use tough::schema::PathSet;
+use tough::{ExpirationEnforcement, FilesystemTransport, Limits, Repository};
+use url::Url;
+
+#[derive(Debug, StructOpt)]
+pub(crate) struct AddRoleArgs {
+    /// Delegating role
+    #[structopt(long = "role", required = true)]
+    role: String,
+
+    /// Delegatee role
+    #[structopt(short = "d", long = "delegatee")]
+    delegatee: String,
+
+    /// Key files to sign with
+    #[structopt(short = "k", long = "key", required = true, parse(try_from_str = parse_key_source))]
+    keys: Vec<Box<dyn KeySource>>,
+
+    /// Expiration of new role file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(short = "e", long = "expires", parse(try_from_str = parse_datetime))]
+    expires: Option<DateTime<Utc>>,
+
+    /// Version of targets.json file
+    #[structopt(short = "v", long = "version")]
+    version: Option<NonZeroU64>,
+
+    /// Path to root.json file for the repository
+    #[structopt(short = "r", long = "root")]
+    root: PathBuf,
+
+    /// TUF repository metadata base URL
+    #[structopt(short = "m", long = "metadata-url")]
+    metadata_base_url: Url,
+
+    /// incoming metadata
```

Comment:
Match casing of other arguments (these are displayed to the user during --help).

```suggestion
    /// Incoming metadata
```